pkg/ipam/crd.go

```diff
@@ -128,12 +128,8 @@ func newNodeStore(nodeName string, conf Configuration, owner Owner, k8sEventReg
 				defer func() { k8sEventReg.K8sEventReceived("CiliumNode", "update", valid, equal) }()
 				if oldNode, ok := oldObj.(*ciliumv2.CiliumNode); ok {
 					if newNode, ok := newObj.(*ciliumv2.CiliumNode); ok {
-						if oldNode.DeepEqual(newNode) {
-							equal = true
-							return
-						}
 						valid = true
-						store.updateLocalNodeResource(newNode.DeepCopy())
+						equal = !store.updateLocalNodeResource(newNode.DeepCopy())
```

Comment:
Overall, I found this change quite hard to understand, but I think a bit of renaming would help. Specifically:
* The name of the function`updateLocalNodeResource` does not indicate that it returns a value, yet here we a receiving a `bool` and inverting it.
*`updateLocalNodeResource` returns a `bool` which it calls `changed`, but here we invert it with `!` and call it `equal`.
* At line 120 in this file, we also call `updateLocalNodeResource` but ignore its return value.

`updateLocalNodeResource` is really doing multiple things: storing the latest node value, determining whether the node's spec or status has changed, and updating ENI routes and allocation pool sizes. I think the code should be refactored to at least move the change detection out of `updateLocalNodeResource`.

Reply:
Good call! I looked at the code again and found that having a side-effect free setter in addition to `updateLocalNodeResource` makes the change much easier to understand.

Please re-review.